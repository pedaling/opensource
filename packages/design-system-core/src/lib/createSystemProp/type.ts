import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
