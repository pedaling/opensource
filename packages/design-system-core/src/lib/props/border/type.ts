import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
