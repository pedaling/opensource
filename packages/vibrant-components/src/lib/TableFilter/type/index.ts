export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
