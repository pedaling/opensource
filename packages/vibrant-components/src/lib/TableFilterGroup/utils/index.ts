import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
