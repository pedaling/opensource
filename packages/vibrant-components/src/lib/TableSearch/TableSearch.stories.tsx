import type { ComponentMeta, ComponentStory } from '@storybook/react';
import type { TableHeader } from '../TableHeader';
import { TableSearch } from './TableSearch';

export default {
  title: 'TableSearch',
  component: TableSearch,
  args: {},
} as ComponentMeta<typeof TableSearch>;

export const Basic: ComponentStory<typeof TableHeader> = props => (
  <TableSearch {...props}>
    <TableSearch.Field placeholder="검색할 단어를 입력하세요." />
  </TableSearch>
);

export const withOption: ComponentStory<typeof TableHeader> = props => (
  <TableSearch {...props}>
    <TableSearch.Option
      options={[
        { label: 'id', value: 'id' },
        { label: 'title', value: 'title' },
        { label: 'content', value: 'content' },
        { label: 'author', value: 'author' },
        { label: 'createdAt', value: 'createdAt' },
      ]}
    />
    <TableSearch.Field placeholder="옵션과 함께 검색할 단어를 입력하세요." />
  </TableSearch>
);
