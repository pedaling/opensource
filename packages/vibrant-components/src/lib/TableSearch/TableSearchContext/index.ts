export { TableSearchProvider, useTableSearch } from './TableSearchContext';
