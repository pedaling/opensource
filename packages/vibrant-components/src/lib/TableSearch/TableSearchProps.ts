import type { ReactElement } from 'react';
import { withVariation } from '@vibrant-ui/core';

export type TableSearchProps = {
  children: ReactElement | ReactElement[];
  testId?: string;
  onSubmit?: (fieldValue: string, optionValue?: string) => void;
  onTextChange?: ({ value, prevent }: { value: string; prevent: () => void }) => void;
};

export const withTableSearchVariation = withVariation<TableSearchProps>('TableSearch')();
