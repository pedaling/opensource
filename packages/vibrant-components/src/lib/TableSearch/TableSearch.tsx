import { useRef, useState } from 'react';
import type { TextInputRef } from '@vibrant-ui/core';
import { HStack } from '../HStack';
import { TableSearchProvider } from './TableSearchContext';
import { TableSearchField } from './TableSearchField';
import { TableSearchOption } from './TableSearchOption';
import type { TableSearchProps } from './TableSearchProps';

export const TableSearch = ({ children, testId = 'table-search', onSubmit }: TableSearchProps) => {
  const [optionValue, setOptionValue] = useState<string>();
  const searchFieldRef = useRef<TextInputRef>(null);

  const submit = (value: string) => {
    onSubmit?.(value, optionValue);
  };

  const changeOption = (value?: string) => {
    setOptionValue(value);

    searchFieldRef.current?.focus();
  };

  return (
    <TableSearchProvider searchFieldRef={searchFieldRef} onOptionChange={changeOption} onSubmit={submit}>
      <HStack width="100%" alignVertical="center" spacing={4} data-testid={testId}>
        {children}
      </HStack>
    </TableSearchProvider>
  );
};

TableSearch.displayName = 'TableSearch';

TableSearch.Field = TableSearchField;

TableSearch.Option = TableSearchOption;
