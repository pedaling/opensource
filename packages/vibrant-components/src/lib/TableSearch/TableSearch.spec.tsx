import { waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { TableSearch } from './TableSearch';

const options = [
  {
    label: 'option 1',
    value: '1',
  },
  {
    label: 'option 2',
    value: '2',
  },
  {
    label: 'option 3',
    value: '3',
  },
];

describe('<TableSearch />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;

  describe('when component is rendering', () => {
    beforeEach(async () => {
      renderer = render(
        <TableSearch>
          <TableSearch.Field />
          <TableSearch.Option options={options} />
        </TableSearch>
      );
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when options change', () => {
    let searchField: HTMLInputElement;

    beforeEach(async () => {
      renderer = render(
        <TableSearch>
          <TableSearch.Field />
          <TableSearch.Option options={options} />
        </TableSearch>
      );

      const selectField = await renderer.findByTestId('select-field');

      await waitFor(() => userEvent.click(selectField.children[1]));

      searchField = (await renderer.findByTestId('table-search-field')) as HTMLInputElement;
    });

    it('TableSearch.Field is focused', () => {
      expect(searchField.id).toEqual(document.activeElement?.id);
    });
  });
});
