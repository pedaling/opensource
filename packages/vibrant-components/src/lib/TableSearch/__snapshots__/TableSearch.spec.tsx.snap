// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableSearch /> when component is rendering match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 9px;
  padding-right: 0;
  width: 100%;
  height: 38px;
  border-width: 1px;
  border-style: solid;
  border-color: #0000000d;
  background-color: #ffffff;
  border-radius: 4px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 20px;
  height: 20px;
  fill: #373737cc;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 100%;
  cursor: text;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: text;
}

.emotion-9 {
  text-align: left;
  color: #373737cc;
  -webkit-text-fill-color: #373737cc;
  text-fill-color: #373737cc;
  position: absolute;
  z-index: 1;
  left: 8px;
  right: 9px;
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.emotion-11 {
  text-align: left;
  color: #373737cc;
  -webkit-text-fill-color: #373737cc;
  text-fill-color: #373737cc;
  font-size: 0.875rem;
  line-height: 1.125rem;
  display: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding-top: 9px;
  padding-left: 8px;
  padding-bottom: 9px;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  background-color: #ffffff00;
  border-width: 0;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  font-size: 0.875rem;
  line-height: 1.125rem;
  width: 100%;
  padding: 0;
  height: 36px;
  padding-top: 9px;
  padding-left: 8px;
  padding-right: 9px;
  padding-bottom: 9px;
}

.emotion-12::-webkit-input-placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-12::-moz-placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-12:-ms-input-placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-12::placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-13 {
  text-align: left;
  color: #373737cc;
  -webkit-text-fill-color: #373737cc;
  text-fill-color: #373737cc;
  font-size: 0.875rem;
  line-height: 1.125rem;
  display: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding-top: 9px;
  padding-right: 9px;
  padding-bottom: 9px;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  width: 4px;
  height: 0;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 120px;
}

.emotion-18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  background-color: #ffffff00;
  border-width: 0;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  font-size: 0.875rem;
  line-height: 1.125rem;
  width: 100%;
  padding: 0;
  position: absolute;
  height: 0;
  opacity: 0;
}

.emotion-19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 38px;
  padding-left: 9px;
  padding-right: 9px;
  border-width: 1px;
  border-style: solid;
  border-color: #0000000d;
  border-radius: 4px;
  background-color: #ffffff;
}

.emotion-21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-right: 8px;
  min-width: 0;
}

.emotion-23 {
  text-align: left;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
  text-transform: none;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-weight: 400;
}

.emotion-24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 20px;
  height: 20px;
  fill: #0c0c0c;
}

<div>
  <div
    class="emotion-0"
    data-testid="table-search"
  >
    <div
      class="emotion-1"
      data-testid="table-search-field"
    >
      <div
        class="emotion-2"
        data-testid="text-field"
      >
        <div
          class="emotion-3"
        >
          <div
            class="emotion-4"
          >
            <svg
              class="emotion-5"
              data-testid="search-thin"
              viewBox="0 0 24 24"
            >
              <path
                class="emotion-6"
                d="m22.45 21.2-5.35-5.35c1.2-1.45 1.9-3.3 1.9-5.35C19 5.8 15.2 2 10.5 2S2 5.8 2 10.5 5.8 19 10.5 19c2.05 0 3.9-.7 5.35-1.9l5.35 5.35c.1.1.25.1.35 0l.9-.9c.1-.1.1-.25 0-.35M3.75 10.5c0-3.7 3.05-6.75 6.75-6.75s6.75 3.05 6.75 6.75-3.05 6.75-6.75 6.75-6.75-3.05-6.75-6.75"
              />
            </svg>
          </div>
          <div
            class="emotion-7"
          >
            <div
              class="emotion-8"
              tabindex="-1"
            >
              <span
                class="emotion-9"
                style="top: 9px; font-size: 0.875rem; line-height: 1.125rem; transition-property: top, fontSize, lineHeight; transition-duration: 100ms; transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);"
              />
            </div>
            <div
              class="emotion-10"
            >
              <span
                class="emotion-11"
              />
              <input
                autocapitalize="none"
                autocomplete="off"
                class="emotion-12"
                inputmode="search"
                type="search"
                value=""
              />
              <span
                class="emotion-13"
              />
            </div>
          </div>
          <div
            class="emotion-4"
          />
        </div>
      </div>
    </div>
    <span
      class="emotion-15"
    />
    <div
      class="emotion-16"
      data-testid="table-search-option"
    >
      <div
        class="emotion-1"
        data-testid="select-field"
      >
        <input
          autocomplete="off"
          class="emotion-18"
          inputmode="text"
          readonly=""
          type="text"
          value="1"
        />
        <div
          class="emotion-19"
          tabindex="0"
        >
          <div
            class="emotion-0"
          >
            <span
              class="emotion-21"
            >
              <span
                class="emotion-6"
              >
                <span
                  class="emotion-23"
                  data-testid="body"
                >
                  option 1
                </span>
              </span>
            </span>
            <svg
              class="emotion-24"
              data-testid="arrowtriangledown-thin"
              viewBox="0 0 24 24"
            >
              <path
                class="emotion-6"
                d="m7.05 9.4 4.75 7c.1.15.3.15.4 0l4.75-7c.1-.15 0-.4-.2-.4h-9.5c-.2 0-.3.25-.2.4"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
